File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int minLen = INT_MAX;
        int len = 0;
        for(int i = 1 ; i < nums.size() ; i++)
        {
            nums[i] = nums[i] + nums[i-1];
        }
        int i = 0;
        int j = 0;
        while(j < nums.size())
        {
            if(nums[j] >= target)
            {
                len = j-i+1;
                minLen = min(len,minLen);
                if(nums[j]-nums[i] >= target) i++;
                else 
                {
                    j++;
                    i = 0;
                }
                
            }
            else
            {
                j++;
            }
        }
        if (len != 0) return minLen;
        else return 0;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int i=0,sum=0,minLen=Integer.MAX_VALUE;
        for (int j=0;j<nums.length;j++){
            sum+=nums[j];
            while(sum>=target){
                minLen = Math.min(minLen,j-i+1);
                sum-=nums[i++];
            }
        }
        if(minLen==Integer.MAX_VALUE){
        return 0;
        }
        else return minLen;
    }
}